enrollment-server-onboarding/src/test/java/com/wultra/app/onboardingserver/impl/service/document/DocumentProcessingServiceTest.java

```diff
@@ -17,58 +17,274 @@
+
+        final DocumentSubmitRequest request = new DocumentSubmitRequest();
+        request.setProcessId("p1");
+        request.setResubmit(true);
+        request.setData("files".getBytes());
+        request.setDocuments(metadata);
+        when(dataExtractionService.extractDocuments(request.getData())).thenReturn(data);
+
+        tested.submitDocuments(identityVerification, request, ownerId);
+        List<DocumentVerificationEntity> documents = documentVerificationRepository.findAll();
+        assertEquals(4, documents.size());
+        assertThat(documents)
+                .extracting(DocumentVerificationEntity::getStatus)
+                .containsOnly(DocumentStatus.VERIFICATION_PENDING, DocumentStatus.DISPOSED);
     }
 
     @Test
-    void pairTwoSidedDocumentsTest() {
-        DocumentVerificationEntity docIdCardFront = new DocumentVerificationEntity();
-        docIdCardFront.setId("1");
-        docIdCardFront.setType(DocumentType.ID_CARD);
-        docIdCardFront.setSide(CardSide.FRONT);
-
-        DocumentVerificationEntity docIdCardBack = new DocumentVerificationEntity();
-        docIdCardBack.setId("2");
-        docIdCardBack.setType(DocumentType.ID_CARD);
-        docIdCardBack.setSide(CardSide.BACK);
-
-        List<DocumentVerificationEntity> documents = List.of(docIdCardFront, docIdCardBack);
-
-        service.pairTwoSidedDocuments(documents);
-        when(documentVerificationRepository.setOtherDocumentSide("1", "2")).thenReturn(1);
-        verify(documentVerificationRepository, times(1)).setOtherDocumentSide("1", "2");
-        when(documentVerificationRepository.setOtherDocumentSide("2", "1")).thenReturn(1);
-        verify(documentVerificationRepository, times(1)).setOtherDocumentSide("2", "1");
-        assertEquals(docIdCardBack.getId(), docIdCardFront.getOtherSideId());
-        assertEquals(docIdCardFront.getId(), docIdCardBack.getOtherSideId());
+    @Sql(scripts = "DocumentProcessingServiceTest.testResubmitDocuments.sql")
+    void testResubmitDocuments_missingOriginalDocumentId() throws Exception {
+        final IdentityVerificationEntity identityVerification = identityVerificationRepository.findById("v1").get();
+        assertNotNull(identityVerification);
+
+        final List<DocumentSubmitRequest.DocumentMetadata> metadata = createIdCardMetadata();
+        final List<Document> data = createIdCardData();
+        final OwnerId ownerId = createOwnerId();
+
+        final DocumentSubmitRequest request = new DocumentSubmitRequest();
+        request.setProcessId("p1");
+        request.setResubmit(true);
+        request.setData("files".getBytes());
+        request.setDocuments(metadata);
+        when(dataExtractionService.extractDocuments(request.getData())).thenReturn(data);
+
+        assertThrows(DocumentSubmitException.class, () -> tested.submitDocuments(identityVerification, request, ownerId));
+    }
+
+    @Test
+    @Sql(scripts = "DocumentProcessingServiceTest.testResubmitDocuments.sql")
```

Comment:
You may also extract a common sql to the class level.

Reply:
hmm, there are three functionalities tested - document pairing, submit and resubmit. Each of which has different starting point. They don't have much in common. Merging them could worse the readability of tests.